src/users/users.service.ts

```diff
@@ -1,33 +1,64 @@
-import { ConflictException, Injectable } from '@nestjs/common'
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common'
 import { PrismaService } from '../prisma/prisma.service'
 import { Prisma, User } from '@prisma/client'
 import { genSalt, hash } from 'bcrypt'
 import { CreateUserInputDto } from './schemas/CreateUserInput.schema'
 import { generateRandomPassword } from './util/passwords.util'
 import { trimMany } from '../shared/util/strings.util'
+import { nanoid } from 'nanoid'
+import { PrismaClientKnownRequestError } from '@prisma/client/runtime/library'
+import { JwtService, TokenExpiredError } from '@nestjs/jwt'
 
-type UniqueUserWhere = Prisma.AtLeast<
-  {
-    id?: number
-    email?: string
-  },
-  'id' | 'email'
->
+const PASSWORD_RESET_TOKEN_TTL = '30 days'
 
 @Injectable()
 export class UsersService {
-  constructor(private prisma: PrismaService) {}
+  constructor(
+    private prisma: PrismaService,
+    private jwt: JwtService,
+  ) {}
 
-  async getAllUsers() {
+  getAllUsers() {
     return this.prisma.user.findMany()
   }
 
-  async findUser(where: UniqueUserWhere): Promise<User | null> {
+  findUser(where: Prisma.UserWhereUniqueInput): Promise<User | null> {
     return this.prisma.user.findUnique({
       where,
     })
   }
 
+  findUserByEmail(email: string) {
+    return this.prisma.user.findFirst({
+      where: { email: { equals: email, mode: 'insensitive' } },
```

Comment:
Oh cool.. I like the use of the `insensitive` mode here.

But we still need to handle the `+` case as well, right?

Typically best to do this kind of normalization/sanitization of user input data on entry to the system, so that everywhere else can just assume that the data is good to act on from there on: 

https://github.com/thegoodparty/gp-api/pull/23#discussion_r1873670011

Reply:
Ah.. you're correct.. I got this mixed up w/ the front-end endpoint receiving the `email` from the URL..  Mah bad.